src/lib/DataTable.tsx

```diff
@@ -259,16 +278,34 @@ class TableHead<T> extends React.Component<TableHeadProps<T>> {
                 data-index={index}
                 colSpan={colSpan}
                 rowSpan={rowSpan}
-                style={column.customStyle && column.customStyle({header: true, name: column.name})}
+                style={style}
                 onClick={this._getOnSortClick(column)}
             >
                 <div className={b('head-cell')}>
                     {header}
                     {<ColumnSortIcon {...column} />}
                 </div>
+                {this.renderCellResizeHandler(column)}
             </th>
         );
     };
+    renderCellResizeHandler(column: HeadCellColumn<T>) {
+        const {onResize} = this.props;
+
+        if (!column.resizeable || !onResize || typeof column.width !== 'number') {
```

Comment:
Is there reasons to not use the condition to check if need to apply `width`/`maxWidth` to `style` of td/th?
I.e. why do we check only `column.resizeable` in line 135/266?

Reply:
I set correct `resizeable` property in `getColumn` method (it also applies default value): https://github.com/gravity-ui/react-data-table/blob/feat/add-columns-resize/src/lib/DataTable.tsx#L1147

So check `if (!column.resizeable)` should be enough in here and everywhere else. However, final `resizeable` property on column doesn't implies actual `onResize` and `width` types. So this check here to not add additional logic to `ResizeHandler` and to not use type assertions.

Reply:
Is it possible to resize a column without explicitly defined `width` property?

Reply:
No. If `width` isn't set explicitly, it will work as before - column width corresponds to content width